src/home/BudgetAtAGlanceWidget.js

```diff
@@ -1,7 +1,245 @@
+        .filter(
+          (transaction) => transaction.date >= firstDay && transaction.date <= lastDay,
+        )
+        .filter(
+          (transaction) => transaction.categoryId !== INCOME_ID,
+        )
+        .reduce(
+          (total, transaction) => total + transaction.amount,
+          0,
+        );
+      return parseFloat(allTransactions);
+    },
+  );
+
+  // Gets categories data, excludes income category, sums target values
+  const totalBudget = useSelector(
+    (state) => {
+      const allCategories = state.categories.entities
+        .filter(
+          (category) => category.id !== INCOME_ID,
+        )
+        .reduce(
+          (total, category) => total + category.target,
+          0,
+        );
+      return parseFloat(allCategories);
+    },
+  );
+
+  // Gets budget percentage to calculate status
+  const spentToBudget = parseFloat(Math.abs(amountSpent) / totalBudget);
+
+  // Gets month percentage to calculate status
+  const currDay = date.getDate();
+  const endDay = lastDay.getDate();
+  const monthPercentage = parseFloat(currDay / endDay);
+
+  let status = '';
+  if (amountSpent >= 0 && spentToBudget <= (monthPercentage - 0.10)) {
+    status = 'Under';
+  } else if (amountSpent >= 0 && spentToBudget > (monthPercentage - 0.10)) {
+    status = 'On Track';
+  } else {
+    if (spentToBudget >= (monthPercentage + 0.10)) {
+      status = 'Over';
+    } else if (spentToBudget <= (monthPercentage - 0.10)) {
+      status = 'Under';
+    } else {
+      status = 'On Track';
+    }
+  }
+
+  return (
+    <div className="row">
+      <div className="col-md-6">
+        <div className="card border-secondary mb-3">
+          <h4 className="card-header">Budget at a Glance</h4>
+          <p className="m-3">
+            <SpentTracker amount={amountSpent} total={totalBudget} />
+            <div className="my-3">
+              <BudgetProgressBar
+                amount={amountSpent}
+                total={totalBudget}
+                ratio={spentToBudget}
+                monthPercent={monthPercentage}
+                status={status}
+              />
+            </div>
+            <div className="my-3">
+              <BudgetStatus
+                status={status}
+              />
+            </div>
+          </p>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Displays amount spent over the total budget.
+ * If credited transactions exceed total budget,
+ * additional funds are displayed in addition to budgeted amount.
+ * Amount is sum of transactions. Total is sum of category targets.
+ *
+ * @component
+ */
+
+function SpentTracker({ amount, total }) {
+  let spentDisplay = '';
+  if (amount > 0) {
+    spentDisplay = (
+      <>
+        $0 spent out of
+        &nbsp;
+        <strike>
+          $
+          {total}
+        </strike>
+        &nbsp;
+        (
+        $
+        {total}
+        +
+        $
+        {amount}
```

Comment:
I recently realize that there's a formatMoney function in the utilities/index.js file. I totally forgot I put that there! It might be helpful here to make sure this money is formatted nicely.